### PR TITLE
Add Build Number into version

### PR DESCRIPTION
**Pull request by hpidcock**
## Description of change

- During make, allow injecting git commit and build number from env.
- Build set on CurrentVersion if the version is a dev build.
- When version tag has a build number, two docker images are tagged and pushed.
- Build number is exposed in version command.

## QA steps

```
make JUJU_BUILD_NUMBER=9001 install
juju version --all
```

## Documentation changes

N/A

## Bug reference

N/A

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -184,29 +188,40 @@ check-deps:
...
+JUJUD_STAGING_DIR          ?= /tmp/jujud-operator
+JUJUD_BIN_DIR              ?= ${GOPATH}/bin
+OPERATOR_IMAGE_BUILD_SRC   ?= true
+OPERATOR_IMAGE_TAG         ?= $(shell jujud version | grep -E -o "^[[:digit:]]{1,9}\.[[:digit:]]{1,9}(\.|-[[:alpha:]]+)[[:digit:]]{1,9}(\.[[:digit:]]{1,9})?")
+# Legacy tags never have a build number.
+OPERATOR_IMAGE_TAG_LEGACY  ?= $(shell jujud version | grep -E -o "^[[:digit:]]{1,9}\.[[:digit:]]{1,9}(\.|-[[:alpha:]]+)[[:digit:]]{1,9}")
+OPERATOR_IMAGE_PATH         = ${DOCKER_USERNAME}/jujud-operator:${OPERATOR_IMAGE_TAG}
+OPERATOR_IMAGE_PATH_LEGACY  = ${DOCKER_USERNAME}/jujud-operator:${OPERATOR_IMAGE_TAG_LEGACY}
+
+check-build:
```
Comment (wallyworld) [inline Makefile]: as discussed irl, rename

File: version/version.go [hunk 2]
```diff
@@ -72,3 +84,22 @@ func IsDev(v semversion.Number) bool {
...
+	return ver
+}
+
+func mustParseBuildInt(buildInt string) int {
+	if buildInt == "" {
+		return 0
+	}
+	i, err := strconv.Atoi(buildInt)
+	if err != nil {
+		panic(err)
```
Comment (tlmiller) [inline version/version.go]: Can you make this error better message better?

**Comment by hpidcock** [comment]
$$merge$$

**Comment by hpidcock** [comment]
[k8s operator build num tag lookup](https://trello.com/c/Ju3QtRJN/919-k8s-operator-build-num-tag-lookup)

**Comment by hpidcock** [comment]
$$merge$$

**Pull request merged**